server/model.js

```diff
@@ -56,13 +56,14 @@ module.exports = {
     try {
       const review = await pool.query(queryStringForReviews, queryParamsForReviews);
       const review_id = review.rows[0].review_id;
-      for (let i = 0; i < photos.length; i++) {
-        await pool.query(queryStringForPhotos, [review_id, photos[i]]);
-      }
+      const queries = [];
+      photos.map((photo) => {
+        queries.push(pool.query(queryStringForPhotos, [review_id, photo]));
+      });
       for (let key in characteristics) {
-        await pool.query(queryStringForCharacteristics,[key, review_id, characteristics[key]]);
+        queries.push(pool.query(queryStringForCharacteristics, [key, review_id, characteristics[key]]));
       }
-      return review.rows[0];
+      return Promise.all(queries);
```

Comment:
@cdeng1111 - Using [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) here will execute all the promises inside the `queries` array asynchronously and return a promise itself. This lets you run the promises in parallel, and then do something after all the promises have been resolved.

It doesn't look like you're using `await` or `.then` with `promise.all`, so JS would continue executing the rest of your code instead of waiting for all the photos to be done inserting.

What if you send the 201 successful response before the photos are done inserting, but then there's a photo insertion error? The successful response was already sent, so it'd be too late to send the error response instead.

I think you would need to use `await` or `.then` with `promise.all` to make sure you don't run `res.send` before the photos are done inserting.

Let me know if that makes sense or you want to chat about it.